src/CompletionHandler.php

```diff
@@ -257,10 +267,18 @@ protected function completeForCommandName()
     {
         if (!$this->command || (count($this->context->getWords()) == 2 && $this->context->getWordIndex() == 1)) {
             $commands = $this->application->all();
-            $names = array_keys($commands);
+            $names = array();
+
+            foreach ($commands as $name => $command) {
+                if ($command->isHidden()) {
+                    continue;
+                }
+
+                if ($name === '_completion') {
```

Comment:
Won't it be better to change `_completion` command declaration to make it hidden as we've discussed? Or there is some idea behind not completing `_completion` command, but still showing it in help command?